fe/fe-core/src/main/java/org/apache/doris/load/update/UpdateStmtExecutor.java

```diff
@@ -177,21 +176,7 @@ private void executePlan() throws Exception {
 
     private void commitAndPublishTxn() throws UserException {
         GlobalTransactionMgr globalTransactionMgr = Catalog.getCurrentGlobalTransactionMgr();
-        // situation1: no data is updated, abort transaction
-        if (effectRows == 0) {
-            LOG.info("abort transaction for update stmt, query id:{}, reason: {}", DebugUtil.printId(queryId),
-                    TransactionCommitFailedException.NO_DATA_TO_LOAD_MSG);
-            globalTransactionMgr.abortTransaction(dbId, txnId, TransactionCommitFailedException.NO_DATA_TO_LOAD_MSG);
-            StringBuilder sb = new StringBuilder();
-            sb.append("{'label':'").append(label);
-            sb.append(", 'txnId':'").append(txnId).append("'");
-            sb.append(", 'queryId':'").append(DebugUtil.printId(queryId)).append("'");
-            sb.append("}");
-            analyzer.getContext().getState().setOk(effectRows, 0, sb.toString());
-            return;
```

Comment:
I remember that Doris do not support commit empty rowset. Have you tested this case yet?

Reply:
yes, i have tested stream load and insert already, it works well. actually, we should keep same behavior with mysql, whether update or insert 0 rows

Reply:
Yes， Doris support empty rowset. Because when there are 10 tablets, but user could insert 1 row, and this row only exist in one tablet and other tablets have empty rowset.

But, we should optimize this. If all tablets have empty rowsets, then just return success, not commit it, and abort the txn.

Reply:
@yiguolei now, commit and publish txn without any tablet  almost cost nothing, we should unity it to avoid user to load data again, because no partition with data load may have some cases for normal and unnormal situation, and we may write extra code to handle this case for insert and update, which make code look like ugly and may not easy to handle

Reply:
Yes, I agree with you.

Then LGTM.